lesson36/src/js/article.js

```diff
@@ -0,0 +1,114 @@
+const articleList = document.getElementById("js-article-List");
+
+const createElementWithClassName = (type, className) => {
+  const element = document.createElement(type);
+  element.className = className;
+  return element;
+};
+
+const createErrorMessage = (text) => {
+  const errorMessage = createElementWithClassName("p", "article-error-message");
+  errorMessage.textContent = text;
+  return errorMessage;
+};
+
+const renderLoading = () => {
+  const loadingWrapper = createElementWithClassName("div", "article-loading-wrapper");
+  const loading = document.createElement("img");
+  loadingWrapper.id = "js-loading";
+  loading.src = "./img/article-loading.gif";
+  articleList.appendChild(loadingWrapper).appendChild(loading);
+};
+
+const removeLoading = () => document.getElementById("js-loading").remove();
+
+const fetchData = async (endpoint) => {
+  const response = await fetch(endpoint);
+  if (!response.ok) {
+    const responseMessage = `${response.status}:${response.statusText}`;
+    articleList.append(createErrorMessage(responseMessage));
+    console.error(responseMessage);
+    return;
+  }
+  return await response.json();
+}
+
+const getArticleData = async (api) => {
+  try {
+    return await fetchData(api);
+  } catch (error) {
+    console.error(error);
+    articleList.before(createErrorMessage(error));
+  }
+}
+
+const renderArticleItems = (articleData) => {
+  articleData.forEach((data) => {
+    const item = createElementWithClassName("li", "article__item");
+    const content = createElementWithClassName("p", "article__item-content");
+    const author = document.createElement("p");
+    author.textContent = data.author;
+    content.textContent = data.quote;
+    articleList.appendChild(item).appendChild(author).after(content);
+  });
+}
+
+let totalPosts;
+const endpointConfig = {
+  path: "https://api.javascripttutorial.net/v1/quotes/",
+  limit: 10,
+  currentPage: 1,
+  setEndpoint: () => {
+    const url = new URL(endpointConfig.path);
+    url.searchParams.set("page", endpointConfig.currentPage);
+    url.searchParams.set("limit", endpointConfig.limit);
+    return url.href;
+  }
+}
+
+const observeConfig = {
+  observe: null,
+  option: { threshold: 1.0 },
+  startObserve: (target, callback) => {
+    observeConfig.observe = new IntersectionObserver(callback, observeConfig.option);
+    observeConfig.observe.observe(target);
+  },
+  stopObserve: (target) => {
+    observeConfig.observe.unobserve(target);
+  }
+}
```

Comment:
I did not know "Intersection Observer "! This is very interesting! I learned a lot!